subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -45,6 +46,120 @@ static void oacp_l2cap_closed(struct bt_gatt_ots_l2cap *l2cap_ctx,
 }
 #endif
 
+#if defined(CONFIG_BT_OTS_OACP_CREATE_SUPPORT)
+static enum bt_gatt_ots_oacp_res_code oacp_create_proc_validate(
+	struct bt_conn *conn,
+	struct bt_ots *ots,
+	struct bt_gatt_ots_oacp_proc *proc)
+{
+	static char * const dflt_name = "";
+
+	char str[BT_UUID_STR_LEN];
+	int err;
+	uint64_t id;
+	struct bt_ots_obj_metadata obj_init;
+	struct bt_gatt_ots_object *obj;
+	struct bt_gatt_ots_oacp_create_params *params = &proc->create_params;
+
+	bt_uuid_to_str(&params->type.uuid, str, BT_UUID_STR_LEN);
+	LOG_DBG("Validating Create procedure with size: 0x%08X and "
+		"type: %s", params->size, log_strdup(str));
+
+	if (!BT_OTS_OACP_GET_FEAT_CREATE(ots->features.oacp)) {
+		LOG_DBG("Create Procedure is not supported.");
+		return BT_GATT_OTS_OACP_RES_OPCODE_NOT_SUP;
+	}
+
+	obj_init.name = dflt_name;
+	obj_init.type = params->type;
+	obj_init.size.cur = 0;
+	obj_init.size.alloc = params->size;
+	obj_init.props = ots->dflt_obj_prop;
+
+	err = bt_ots_obj_add(ots, &obj_init, NULL, &id);
+	if (err) {
+		LOG_ERR("Adding object during Create procedure failed: %d", err);
+		goto exit;
+	}
+
+	err = bt_gatt_ots_obj_manager_obj_get(ots->obj_manager, id, &obj);
+	if (err) {
+		LOG_ERR("Could not find added object during Create procedure: %d", err);
+		goto exit;
+	}
+
+	__ASSERT(BT_OTS_OBJ_GET_PROP_WRITE(obj->metadata.props),
+		 "Created object must have write property");
+
+	if (IS_ENABLED(CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT)) {
```

Comment:
>@ekohandel You write:
>>Giving a non-empty name would violate the requirements in 3.3.2.1.1 Initialization of Object Metadata.

>You cannot give a name when creating an object (the create procedure is just given a size and a UUID, right?)
>You have to create the object, and then change the name.

That is correct, the create procedure is expected to leave the object with an empty name. My comment was regarding your earlier comment when you suggested:
>In that case the server shall not only update the pointer, but also give it a non-empty name to avoid deletion on disconnect (the latter still remains unimplemented).

I thought your comment had suggested that the server would provide a non-empty name to the object upon creation to prevent it from being deleted. That is only appropriate if the object is being created locally and not by a remote client per requirement in `3.3.2.1.1 Initialization of Object Metadata`.

I think adding the `BT_OTS_OACP_CREATE_SUPPORT` dependency on  `BT_OTS_OBJ_NAME_WRITE_SUPPORT` either via `depends on` or `select` will create a better user experience as you have suggested.